src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -256,19 +266,33 @@ public String produceDts(Compiler compiler, Depgraph depgraph) {
       // parent symbol.
       if (symbol.getName().contains(".prototype")) continue;
 
-      String namespace = getNamespace(symbol.getName());
+      String parentPath = getNamespace(symbol.getName());
       boolean isDefault = isDefaultExport(symbol);
-      boolean isTypedef = symbol.getType() != null && isTypedef(symbol.getType());
-      // Typedef symbols do not enumerate their namespace properties.
-      if (!isDefault && !isTypedef) visitedNamespaces.add(symbol.getName());
-      if (isDefault && visitedNamespaces.contains(namespace)) continue;
-      declareNamespace(isDefault ? namespace : symbol.getName(), symbol, isDefault, compiler, transitiveProvides, true);
+
+      if (visitedNamespaces.contains(parentPath)) {
+        // before we continue we need to account for the fact that the class has been visited.
```

Comment:
Maybe formulate the comment as `// Remember that this class has been visited before continuing.`?